modules/enterprise/server/jar/src/main/java/org/rhq/enterprise/server/rest/SetCallerInterceptor.java

```diff
@@ -72,6 +81,14 @@ public Object setCaller(InvocationContext ctx) throws Exception {
 
         Subject caller=null;
         java.security.Principal p = ejbContext.getCallerPrincipal();
+        if (!startupBean.isInitialized()) {
+            String notInitMessage = "Tried to call REST endpoint but the server has not finished to startup";
+            log.debug(notInitMessage);
+            Response.ResponseBuilder builder;
```

Comment:
Combine lines 87 and 88.

On another note, should we set a retry header here like we do in the AuthenticationFilter on legacy UI?